protocol/x/clob/memclob/memclob.go

```diff
@@ -1352,18 +1353,23 @@ func (m *MemClobPriceTimePriority) validateNewOrder(
 		return types.ErrInvalidReplacement
 	}
 
-	// If the order is a reduce-only order, we should ensure the full size of the order does not increase the
-	// subaccount's current position size. Note that we intentionally do not validate that the reduce-only order
-	// does not change the subaccount's position _side_, and that will be validated if the order is matched.
-	// TODO(DEC-1228): use `MustValidateReduceOnlyOrder` and move this to `PerformStatefulOrderValidation`.
+	// If the order is a reduce-only order, we should ensure that the sign of the order size is the opposite of
+	// the current position size. Note that we do not validate the size/quantity of the reduce only order fill,
+	// as that will be validated if the order is matched.
+	// The subaccount's current position size is defined as is the current state size + any partial fills
+	// that might have occured as a result of this reduce only order replacing another partially filled order.
 	if order.IsReduceOnly() {
-		// If there exists an non-reduce only order with the same order id
-		// cancel the incoming reduce only order by returning an error.
-		if restingOrderExists && !existingRestingOrder.IsReduceOnly() {
-			return types.ErrReduceOnlyOrderReplacement
+		existingPositionSize := m.clobKeeper.GetStatePosition(ctx, orderId.SubaccountId, order.GetClobPairId())
+		existingOrderFillAmountExists, existingOrderFillAmount, _ := m.clobKeeper.GetOrderFillAmount(ctx, orderId)
+		// If there exists an pre existing order fill, add it to position size in state to get current position size.
+		if existingOrderFillAmountExists && restingOrderExists {
```

Comment:
Can someone double check this logic/think through it?
```
	existingRestingOrder, restingOrderExists := m.openOrders.getOrder(ctx, orderId)
	existingMatchedOrder, matchedOrderExists := m.operationsToPropose.MatchedOrderIdToOrder[orderId]
```
There is resting order and matched order. In order to find out what the SIDE of the original order being replaced was, I check resting order. Is there any way for the resting order and matched order to be different sides? Should I be checking the matched order side as well?

Reply:
why doesn't the existing position size in state match the current position size?